src/Core/Resolvers/SqlQueryEngine.cs

```diff
@@ -272,17 +271,19 @@ public object ResolveList(JsonElement array, IObjectField fieldSchema, ref IMeta
             }
             else if (array.ValueKind is JsonValueKind.String)
             {
-                using ArrayPoolWriter buffer = new();
-
-                string text = array.GetString()!;
-                int neededCapacity = Encoding.UTF8.GetMaxByteCount(text.Length);
-                int written = Encoding.UTF8.GetBytes(text, buffer.GetSpan(neededCapacity));
-                buffer.Advance(written);
-
-                Utf8JsonReader reader = new(buffer.GetWrittenSpan());
-                foreach (JsonElement element in JsonElement.ParseValue(ref reader).EnumerateArray())
+                string jsonString = array.GetString()!;
+                if (!string.IsNullOrEmpty(jsonString))
                 {
-                    resolvedList.Add(element);
+                    using (JsonDocument jsonDocument = JsonDocument.Parse(jsonString))
```

Comment:
Will this JsonDocument be properly disposed of?

Reply:
yeah, wondering if have you noticed any issues regarding this in the past?

Reply:
we had some issues with json documents leading to memory leaks, but the docs say this is how to use it

https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/use-dom#use-jsondocument


> Creates the JsonDocument instance in a [using statement](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/using) because JsonDocument implements IDisposable. After a JsonDocument instance is disposed, you lose access to all of its JsonElement instances also. To retain access to a JsonElement instance, make a copy of it before the parent JsonDocument instance is disposed. To make a copy, call [JsonElement.Clone](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonelement.clone). For more information, see [JsonDocument is IDisposable](https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/migrate-from-newtonsoft#jsondocument-is-idisposable).